[Wu Xingyu] iP

DavinciDelta opened the pull request.

DavinciDelta committed:
Add Increment: Level-0

Renamed the project and added in greeting and exit statements

DavinciDelta committed:
Add Increments: Level-1

Allow chatbot to echo user input, provide characteristics to chatbot

DavinciDelta committed:
Update Duke.java

Changed the name of the class from Duke to Chatbot for easier reference and seperated out the process function to another class

DavinciDelta committed:
Add Increments: Level-2

Added new class Task, added a list to ResponseProcessor and new functions to it

DavinciDelta committed:
Add Increments Level-3: Mark as Done

Added the mark and unmark functions to the chatbot

Included checks for number validity

DavinciDelta committed:
Updata Task.java

danielpappa commented on src/main/java/ResponseProcessor.java:
```diff
@@ -0,0 +1,54 @@
+public class ResponseProcessor {
+    List<Task> taskList = new ArrayList<>();
+    public void process(String response) {
+        if ("list".equalsIgnoreCase(response)) {
```
Consider simplifying this code by using a switch statement

danielpappa commented on src/main/java/ResponseProcessor.java:
```diff
@@ -0,0 +1,54 @@
+            String value = response.split(" ")[1];
+            if (isInt(value)) {
+                int number = Integer.parseInt(value) - 1;
+                if ((0 <= number) && (number < taskList.size())) {
```
This if loop is very deeply nested and I would also recommend using a single boolean value with a clear text

danielpappa commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,34 @@
+    private boolean isCompleted;
+
+    public Task(String description, boolean isCompleted) {
+        setDescription(description);
```
Nice way of handling the generator, good abstraction level

danielpappa commented on src/main/java/ResponseProcessor.java:
```diff
@@ -0,0 +1,54 @@
+        } else if (response.toLowerCase().startsWith("mark".toLowerCase())) {
+            String value = response.split(" ")[1];
+            if (isInt(value)) {
+                int number = Integer.parseInt(value) - 1;
```
Consider changing both the variable name of value and number to add more meaning to the variable;

danielpappa reviewed (commented):
Pretty neat and concise code

DavinciDelta committed:
Add Increments: Level-4

Implement ToDos, Events and Deadline classes

Segregate the response processing and task processing

DavinciDelta committed:
Improve code structure

Added hashmap for commands, seperate each command into a different class

DavinciDelta committed:
Add Increments: Level-5

Added more error handling and input output txt files

DavinciDelta committed:
Merge pull request #1 from DavinciDelta/branch-Level-5

Add Increments: Level-5

Musfirahe0556596 commented on src/main/java/main/ResponseProcessor.java:
```diff
@@ -0,0 +1,47 @@
+import java.util.HashMap;
+import java.util.List;
+public class ResponseProcessor {
+    public List<Task> taskList = new ArrayList<>();
```
Plural form should be used on names representing a collection of objects.

Musfirahe0556596 commented on src/main/java/commands/EventCommand.java:
```diff
@@ -0,0 +1,26 @@
+        int indexFrom = getIndex(statement, "/from");
+        int indexTo = getIndex(statement, "/to");
+        String description = statement.substring(0, indexFrom);
+        String from = statement.substring(indexFrom + 5, indexTo);
+        String to = statement.substring(indexTo + 3);
```
Avoid magic numbers e.g. +5, can put them as named constants

Musfirahe0556596 commented on src/main/java/main/ResponseProcessor.java:
```diff
@@ -0,0 +1,47 @@
+        put("list", new ListCommand());
+    }};
+
+    public void process(String response) {
```
Method name could be named better e.g. what does it process?

Musfirahe0556596 reviewed (commented):
Overall great job! Some coding quality and standards to follow!

DavinciDelta committed:
Add Increments: Level-6

Added the delete function into the chatbot

Included enable and disable ascii art command

DavinciDelta committed:
Add Increments: Level-7

Added the save and load file classes and functions

DavinciDelta committed:
Merge branch 'branch-Level-7'

DavinciDelta committed:
Add Increments: A-Jar

Packaged the app as an executable JAR file

DavinciDelta committed:
Add Increments: Level-8

Added the date time parsing if format matches

Changed the ascii art handler to include "picture"

DavinciDelta committed:
Add Increments: Level-9

Include the find command to check for words in list

DavinciDelta committed:
Add Increment: A-JavaDoc

Add Java documentations to the code for most classes and functions

DavinciDelta committed:
Merge pull request #2 from DavinciDelta/branch-Level-8

Add Increments: Level-8

DavinciDelta committed:
Merge pull request #3 from DavinciDelta/branch-Level-9

Add Increments: Level-9

DavinciDelta committed:
Merge branch 'master' into branch-A-JavaDoc

DavinciDelta committed:
Merge pull request #4 from DavinciDelta/branch-A-JavaDoc

Add Increment: A-JavaDoc

DavinciDelta committed:
Update README.md

change theme

DavinciDelta committed:
Revert "Update README.md"

This reverts commit 2af10c3ba40512393eb37b83b6c9e4905f88922f.

DavinciDelta committed:
add remote theme

add a remote theme to the page

DavinciDelta committed:
Add increment: A-UserGuide

Include a user guide for the program

DavinciDelta committed:
Add help function

Included the help function to the code

DavinciDelta committed:
Update ReadME and help

Updated readme and help function to provide more informations

DavinciDelta committed:
Update ResponseProcessor to parser

Improve code clarity issue

Improve the help function to fit command prompt better

DavinciDelta committed:
Update README.md

Improve the description

DavinciDelta committed:
Update ascii art

Fix the ascii art issue printing question marks in command line, improve ascii art quality too

DavinciDelta committed:
Update README.md

DavinciDelta committed:
Update README.md

DavinciDelta committed:
Update AsciiArt.java

Fix minor bug
